src/http/server/response.cr

```diff
@@ -40,6 +40,26 @@ class HTTP::Server
       output.response = self
     end
 
+    def remote_address
+      case @io
```

Comment:
This entire method could be much simpler:

```cr
io = @io
if io.is_a?(OpenSSL::SSLSocket)
  io = io.@bio.io
end

if io.responds_to?(:remote_address)
  io.remote_address
end
```

It's probably fine to just return `nil` in case `remote_address` is not available. Or we could have both a raising and nilable variant.

Reply:
Maybe it would even be better to delegate `remote_address` on `SSLSocket`.

Reply:
So a `remote_address?` and `remote_address`?

Reply:
The reason for `case` is for extending it to support any other wrappers for socket, although I think SSLSocket is the only one so I don't mind changing it.

Reply:
Yes. But I'm not sure if we need the raising variant.

Reply:
We want the raising variant. The underlying IO for a HTTP server should always be a Socket, so we don't have to deal with a nilable. Just like `array[index]` is expected to return a value, `remote_address` is expected to return an address.